astropy/io/fits/tests/test_header.py

```diff
@@ -2961,6 +2961,16 @@ def test_fitsheader_script(self):
                                                  compressed=True)
         hf.close()
 
+    def test_fitsheader_compressed_from_primary_image_ext(self):
+        """Regression test for issue https://github.com/astropy/astropy/issues/7312"""
+        data = np.arange(100*100, dtype=np.float64).reshape((100, 100))
```

Comment:
Do we really need a 100 x 100 float64 to test a header? A 2 x 2 int8 would work just as well?

Reply:
Right! Sorry!

Reply:
No need to apologize and thanks for fixing an old bug! 😸

Reply:
And I forgot the int8 modification... I'm missing caffeine apparently today... :)